redis/src/main/scala/zio/redis/TestExecutor.scala

```diff
@@ -2394,6 +2409,32 @@ private[redis] final class TestExecutor private (
   ): USTM[RespValue] =
     STM.ifM(predicate)(program, STM.succeedNow(Replies.Error))
 
+  private[this] def apply[A, B](optionAB: Option[A => B])(optionA: Option[A]): Option[B] = for {
+    a  <- optionA
+    ab <- optionAB
+  } yield ab(a)
+
+  private[this] def orMissingParameter2[A, B](paramA: Option[A], paramB: Option[B])(
+    program: (A, B) => USTM[RespValue]
+  ): USTM[RespValue] =
+    apply(apply(Some(program.curried))(paramA))(paramB).getOrElse(STM.succeedNow(Replies.Error))
+
+  private[this] def orMissingParameter3[A, B, C](paramA: Option[A], paramB: Option[B], paramC: Option[C])(
+    program: (A, B, C) => USTM[RespValue]
+  ): USTM[RespValue] =
+    apply(apply(apply(Some(program.curried))(paramA))(paramB))(paramC).getOrElse(STM.succeedNow(Replies.Error))
+
+  private[this] def orMissingParameter4[A, B, C, D](
+    paramA: Option[A],
+    paramB: Option[B],
+    paramC: Option[C],
+    paramD: Option[D]
+  )(
+    program: (A, B, C, D) => USTM[RespValue]
+  ): USTM[RespValue] =
+    apply(apply(apply(apply(Some(program.curried))(paramA))(paramB))(paramC))(paramD)
+      .getOrElse(STM.succeedNow(Replies.Error))
+
```

Comment:
As discussed, we can later replace this apply reimplementation with the existing stuff present in zio-prelude.